benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadServer.java

```diff
@@ -0,0 +1,225 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.benchmarks.driver;
+
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.Status;
+import io.grpc.benchmarks.ByteBufOutputMarshaller;
+import io.grpc.benchmarks.Utils;
+import io.grpc.benchmarks.proto.BenchmarkServiceGrpc;
+import io.grpc.benchmarks.proto.Control;
+import io.grpc.benchmarks.proto.Messages;
+import io.grpc.benchmarks.proto.Stats;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.TestUtils;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+import java.io.File;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implements the server-side contract for the load testing scenarios.
+ */
+class LoadServer {
+
+  /**
+   * Generic version of the streaming ping-pong method call.
+   */
+  static MethodDescriptor<ByteBuf, ByteBuf> GENERIC_STREAMING_PING_PONG_METHOD =
+      MethodDescriptor.create(
+          BenchmarkServiceGrpc.METHOD_STREAMING_CALL.getType(),
+          BenchmarkServiceGrpc.METHOD_STREAMING_CALL.getFullMethodName(),
+          new ByteBufOutputMarshaller(),
+          new ByteBufOutputMarshaller());
+
+  private static final Logger LOG = Logger.getLogger(LoadServer.class.getName());
+
+  private final Server server;
+  private final BenchmarkServiceImpl benchmarkService;
+  private final AtomicBoolean shutdown = new AtomicBoolean();
+  private final int port;
+  private ByteBuf genericResponse;
+  private long lastStatTime;
+
+  LoadServer(Control.ServerConfig config) throws Exception {
+    LOG.log(Level.INFO, "Server Config \n" + config.toString());
+    port = config.getPort() ==  0 ? TestUtils.pickUnusedPort() : config.getPort();
+    ServerBuilder<?> serverBuilder = ServerBuilder.forPort(port);
+    switch (config.getServerType()) {
+      case ASYNC_SERVER: {
+        serverBuilder.executor(Executors.newFixedThreadPool(config.getAsyncServerThreads(),
```

Comment:
Direct executor is a special case here. I've asked Vijay to allow the load driver to control running server 'variants' and the server would return a description of the variant in the interface so the driver could report it properly.

The fundamental issue we have here is that the notions of sync/asyc/blocking etc are not that similar between the C & Java runtimes